prov/util/src/util_coll.c

```diff
@@ -330,7 +323,7 @@ static inline uint64_t util_coll_get_next_tag(struct util_coll_mc *coll_mc)
 
 	tag = tag << 16 | coll_mc->cid;
 	tag = tag << 16 | coll_mc->tag_seq++;
-	return tag;
+	return tag | OFI_TAG_COLL;
```

Comment:
I realize this is existing code, but it might probably be cleaner if collective tag (id?) were defined as a u32 and only used in that way.  Rename OFI_TAG_COLL to OFI_COLL_TAG_FLAG.

The use of tag to mean different things is getting confusing.  Particularly, when the entire tag is formed in separate locations.